### PR TITLE
Python: Add method to get IPF path

### PR DESCRIPTION
**Pull request by DavidFair**
**Description of work.**
Moves some code out of `LoadInstrument` into a module where anyone can call it to use IPF files.
Each commit is stand-alone and the work breaks down as follows:

- Move loop into IPF loading, so it will check each instrument dir itself rather than being told to
- Move IPF loading into Experiment Info and add tests
- Move various static file handling out of the ever-growing `ExperimentInfo` into their own file
- Add Python Export
- Rename the new module to `FileFinderHelpers` (although I'm up for any other name suggestions)

**To test:**
- Fire up mantid and check this gives a sane result:

```python
import mantid.api
mantid.api.FileFinderHelpers.getIPFPath("GEM")  # Should return the path to `GEM_Parameters.xml`
mantid.api.FileFinderHelpers.getIPFPath("NotThere") # Should return nothing
```

*There is no associated issue.* - Was required for TOML work in #28513 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/API/inc/MantidAPI/FileFinderHelpers.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#pragma once
+
+#include "MantidAPI/DllConfig.h"
+
+#include <string>
+#include <vector>
+
+namespace Mantid::API {
+
+class MANTID_API_DLL FileFinderHelpers {
```
Comment (martyngigg) [inline Framework/API/inc/MantidAPI/FileFinderHelpers.h:16]: Good refactoring with this. There was no reason this stuff needed to be on an instance.

As you say I'm not super-keen on the name. Everything is instrument-related, right? Some ideas:

- `InstrumentFiles` -> `InstrumentFiles::getInstrumentFilename`
- `InstrumentFileHelpers` -> `InstrumentFileHelpers::getInstrumentFilename`

?

File: Framework/API/inc/MantidAPI/FileFinderHelpers.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+public:
+  /// Search instrument directories for Parameter file,
+  /// return full path name if found, else "".
+  static std::string getIPFPath(const std::string &instName,
```
Comment (martyngigg) [inline Framework/API/inc/MantidAPI/FileFinderHelpers.h:20]: Just a question on symmetry. We have `getInstrumentFilename` below but then an abbreviated `getIPFPath` so we've got both path and filename. If we did go with having Instrument in class name then how about `getParamterFilename()` for this?

File: Framework/PythonInterface/mantid/api/src/Exports/ExperimentInfo.cpp [hunk 3]
```diff
@@ -80,7 +86,7 @@ void export_ExperimentInfo() {
            "file will encapsulate this date (e.g '1900-01-31 23:59:00')\n"
            "\nreturns : list of absolute paths for each valid file\n")
       .staticmethod("getResourceFilenames")
-      .def("getInstrumentFilename", &ExperimentInfo::getInstrumentFilename,
+      .def("getInstrumentFilename", &FileFinderHelpers::getInstrumentFilename,
```
Comment (martyngigg) [inline Framework/PythonInterface/mantid/api/src/Exports/ExperimentInfo.cpp:89]: Should we consider adding a deprecation warning to this pointing to the new class? We can then remove it in the future?

**Review by martyngigg — CHANGES_REQUESTED** [review]
Nice separation of existing functionality into a reusable component. Just a couple of minor questions around naming.

File: Framework/API/inc/MantidAPI/FileFinderHelpers.h (hunk 2 above)
Comment (DavidFair) [reply Framework/API/inc/MantidAPI/FileFinderHelpers.h:20]: I've gone for `getParameterPath` as this returns a full path
`getInstrumentFilename` should be really called `getInstrumentDefPath` or something similar, but for backwards compatibility and as everyone knows it as the former I've left it as is

**Review by martyngigg — APPROVED** [review]
Looks good to me now. :+1:

**Pull request merged**